### PR TITLE
ignore invalid coverage measures

### PR DESCRIPTION
**Pull request by jmecosta**
@guwirth @Bertk  lets ignore the coverage measures?

### CODE REVIEW

**Comment by guwirth** [comment]
@jmecosta looks good to me, closes #842.
Later on I would add a config setting to decide continue/break (#758)

**Comment by jmecosta** [comment]
@guwirth i will create the flag now, and make it disable by default.

**Comment by jmecosta** [comment]
should now cover #842 and #758

**Comment by guwirth** [comment]
@jmecosta I would not introduce a switch to the coverage sensor, I would introduce a general switch, see my last proposal in #758. Doing it as you did it would mean in consequence that we have to (can) add settings to each sensor individual. Not sure if these are not too much choices for a normal user?

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java [hunk 1]
```diff
@@ -163,8 +164,16 @@ private void saveMeasures(SensorContext context,
           }                             
         }
         
-        newCoverage.save();
-        
+        try
+        {
+          newCoverage.save();
+        } catch(Exception ex) {
+          LOG.error("Cannot save measure '{}' : '{}', ignoring measure", filePath, ex.getMessage());
```
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: @jmecosta setting name and flag name do not match? ignoreInvalidMeasures and FAIL_ON_INVALID_MEASURES have invers logic?

Comment (jmecosta) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: yep, forgot to change that. now done

**Comment by guwirth** [comment]
@bertk we need a third opinion ;-). One setting for all sensors to ignore invalid input or individual settings for each sensor?
@all opinions from others are also welcome.

**Comment by Bertk** [comment]
I am fine to ignore invalid input data as long we have error messages for this.

**Comment by guwirth** [comment]
@Bertk question was, use
* `sonar.cxx.coverage.failOnInvalidMeasures` for coverage sensor only
* or one global setting `sonar.cxx.errorRecoveryEnabled` to put all sensor in a tolerant mode

**Comment by Bertk** [comment]
@guwirth 
I missed this detail - sorry :grin:
I think the analysis should not stop if a sensor has an issue e.g. invalid or outdated data. Until now I observed the wrong line number issue with code coverage files only. The code coverage reports might be created on special test systems and not on the build systems which execute the static code analysis.
We should have a global setting for the _tolerant mode_. This is also easier to handle by users and it should be the default.

**Comment by guwirth** [comment]
@jmecosta hi, what should we do with this now?  I still would prefer to have only one setting to put the whole plugin into a tolerant mode.

**Comment by jmecosta** [comment]
Yep, I will make the change soon. As per vote

**Comment by jmecosta** [comment]
should be ready

**Comment by guwirth** [comment]
@jmecosta thx for fixing this.

**Pull request merged**